python/cudf_polars/cudf_polars/dsl/expressions/struct.py

```diff
@@ -0,0 +1,134 @@
+                # Failed to unpack string
+                function = None  # pragma: no cover
+            if function != "StructFunction":
+                raise ValueError("StructFunction required")  # pragma: no cover
+            return getattr(cls, name)
+
+    __slots__ = ("name", "options")
+    _non_child = ("dtype", "name", "options")
+
+    _valid_ops: ClassVar[set[Name]] = {
+        Name.FieldByIndex,
+        Name.FieldByName,
+        Name.RenameFields,
+        Name.PrefixFields,
+        Name.SuffixFields,
+        Name.JsonEncode,
+    }
+
+    def __init__(
+        self,
+        dtype: DataType,
+        name: StructFunction.Name,
+        options: tuple[Any, ...],
+        *children: Expr,
+    ) -> None:
+        self.dtype = dtype
+        self.options = options
+        self.name = name
+        self.children = children
+        self.is_pointwise = True
+        if self.name not in self._valid_ops:
+            raise NotImplementedError(
+                f"Struct function {self.name}"
+            )  # pragma: no cover
+
+    def do_evaluate(
+        self, df: DataFrame, *, context: ExecutionContext = ExecutionContext.FRAME
+    ) -> Column:
+        """Evaluate this expression given a dataframe for context."""
+        columns = [child.evaluate(df, context=context) for child in self.children]
+        (column,) = columns
+        if self.name == StructFunction.Name.FieldByName:
+            field_index = next(
+                (
+                    i
+                    for i, field in enumerate(self.children[0].dtype.polars.fields)
+                    if field.name == self.options[0]
+                ),
+                None,
+            )
+            assert field_index is not None
+            return Column(
+                column.obj.children()[field_index],
+                dtype=self.dtype,
+            )
+        elif self.name == StructFunction.Name.JsonEncode:
+            buff = StringIO()
+            target = plc.io.SinkInfo([buff])
+            table = plc.Table(column.obj.children())
+            metadata = plc.io.TableWithMetadata(
+                table,
+                [(field.name, []) for field in self.children[0].dtype.polars.fields],
+            )
+            options = (
+                plc.io.json.JsonWriterOptions.builder(target, table)
+                .lines(val=True)
+                .na_rep("null")
+                .include_nulls(val=True)
+                .metadata(metadata)
+                .utf8_escaped(val=False)
+                .build()
+            )
+            plc.io.json.write_json(options)
+            return Column(
+                plc.Column.from_iterable_of_py(buff.getvalue().split()),
+                dtype=self.dtype,
+            )
+        elif self.name in {
+            StructFunction.Name.RenameFields,
+            StructFunction.Name.PrefixFields,
+            StructFunction.Name.SuffixFields,
+        }:
+            return column
```

Comment:
For my own understanding: how can we just return `column` here without doing anything (beyond `child.evaluate`) to it? I guess that takes care of everything.